src/db/documentStore.ts

```diff
@@ -0,0 +1,81 @@
+import { DynamoDB } from "@aws-sdk/client-dynamodb";
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  GetCommand,
+  DeleteCommand,
+  QueryCommand,
+} from "@aws-sdk/lib-dynamodb";
+import { MatchDetails } from "../lib/crawler/crawler";
+
+const dynamoDBClient = new DynamoDB({ region: "us-east-1" });
+const dynamoDBDocClient = DynamoDBDocumentClient.from(dynamoDBClient);
+
+type DocumentType = "match_details" | "character_meta" | "crawler_state";
+
+class DocumentStore<T extends MatchDetails> {
+  private name: string;
+  private documentType: DocumentType;
+
+  constructor(tableName: string, documentType: DocumentType) {
+    this.name = tableName;
+    this.documentType = documentType;
+  }
+
+  // Create
+  async update(item: T): Promise<T> {
+    const params = {
+      TableName: this.name,
+      Item: item,
+    };
+    await dynamoDBDocClient.send(new PutCommand(params));
+    return item;
+  }
+
+  // Read
+  async get(key: Record<string, unknown>): Promise<T | undefined> {
```

Comment:
I was hoping for typesafety with they parameters here instead of unknown, but I suppose this will work for now.